### DISCUSSION TITLE
Shuffler Outlook-exchange app

### DISCUSSION THREAD

**Discussion by mirceacroitoru** [Shuffle/Shuffle#1076] [Q&A] [answered]
Hi

I am trying to send an email using Outlook-exchange app. Its a basic workflow with just a webhook and a shuffler tool repeat back to me.
I am being told to use Use the Outlook Office365 app to connect to Office365. Basic auth is deprecated.
I am introducing the App Password but i am getting same message.

I used this in the past with basic auth however i am not able to make it work with App pass.
Any guidelines? Another way of sending emails?

**Comment by frikky** [comment, marked as answer]
Hey,

Please use this app by downloading and activating it (download top right corner): https://shuffler.io/apps/outlook
If you go to your workflow on the left side and search for "office365" it should also show up, and will be automatically enabled for you. The system is built this way as we can't load in 2500~ apps all at once to your instance, so in certain cases, you'll need to activate them yourself.

Hope this helps :)

- Fred

**Reply by mirceacroitoru** [reply]
Hi frikky
I have shuffle installed on local linux machine. I can't find how to download the app.
registering on shuffle website i am able to find it but i need it locally.
Its not on https://github.com/Shuffle/python-apps

**Reply by frikky** [reply]
Click the download button, then upload it in your local instance's app page (drag and drop it in /apps)! This is a workaround that works without internet as well, as long as you can get a hold of the app :)

![image](https://user-images.githubusercontent.com/5719530/229105110-abb75d7f-2515-4da4-96d4-f1549bbcc4a9.png)

**Reply by mirceacroitoru** [reply]
very nice workaround! it worked
Thank you frikky

**Reply by mirceacroitoru** [reply]
Hi frikky

So i configured the app also doing all the things in O365.
 I am trying to configure the app to send emails.
Authentication looks ok, however do not receive email.

I modified the app according to the guidelines. 

![image](https://user-images.githubusercontent.com/104731554/230877187-bb4270e0-61d1-4668-b91b-7784a949dfde.png)


App is configured like this :

Authentication: Mail.Send offline_access ( 1.1.0)
Find Actions : Send mail

Headers,Queries as default
Ssl verify : False
To file : False

Subject and Content i just put random data
Target : email recipient
Cc : empty.

In the execution details i have following 

"Results for Outlook_Office365_1":{6 items
"success":true
"status":405
"url":"https://IP:3443//set_authentication/v1.0/me/sendMail"
"headers":{5 items
"Server":"nginx"
"Date":"Mon, 10 Apr 2023 09:26:30 GMT"
"Content-Type":"text/html"
"Content-Length":"150"
"Connection":"keep-alive"
}
"body":"<html> <head><title>405 Not Allowed</title></head> <body> <center><h1>405 Not Allowed</h1></center> <hr><center>nginx</center> </body> </html> "
"cookies":{0 items
}
}


Any idea how to troubleshoot this?

**Reply by mirceacroitoru** [reply]
Hei
Anyone that can help with this?
